### PR TITLE
json encoder

### PR DESCRIPTION
**Pull request by idoshr**
Fix issue while try to inherit from MongoEngineJSONEncoder
and for bonus add doc of how add encoder hope it will be helpful

### CODE REVIEW

File: flask_mongoengine/json.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def default(obj):
                 (BaseDocument, QuerySet, CommandCursor, DBRef, ObjectId),
             ):
                 return _convert_mongo_objects(obj)
-            return super().default(obj)
+            return superclass.default(obj)
```
Comment (insspb) [inline flask_mongoengine/json.py:81]: What? Where it from?

File: flask_mongoengine/exceptions.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from werkzeug.exceptions import HTTPException
```
Comment (insspb) [inline flask_mongoengine/exceptions.py]: Please revert this exception and all related changes. Leave it to flask.

File: docs/custom_json_encoder.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# Custom Json Encoder
+
+flask-mongoengine have option to add custom encoder for flask
+By this way you can handel encoding special object
```
Comment (insspb) [inline docs/custom_json_encoder.md]:
```suggestion
By this way you can handle encoding special object
```

File: flask_mongoengine/pagination.py [hunk 4]
```diff
@@ -1,37 +1,38 @@
...
 from mongoengine.queryset import QuerySet
 
+from flask_mongoengine.exceptions import InvalidPage
+
 __all__ = ("Pagination", "ListFieldPagination")
 
 
 class Pagination(object):
-    def __init__(self, iterable, page, per_page):
+    def __init__(self, iterable, page, per_page, max_depth=None):
```
Comment (insspb) [inline flask_mongoengine/pagination.py]: If you are working with class, please create docstrings with parameters description. For new and old.

**Review by insspb — CHANGES_REQUESTED** [review]
@idoshr thank you for interest to this project. 

I reviewed changes, being out of mongo right now. Please fix mentioned errors and I will recheck again.

**Comment by idoshr** [comment]
@insspb 
What is your opinion on adding this default encoder
image

**Comment by EvanBldy** [comment]
Hi that's cool ! Thanks
Will you do a new release on pypi after that ?

**Comment by idoshr** [comment]
This pull request spliced to 3 separated PRs
https://github.com/MongoEngine/flask-mongoengine/pull/526
https://github.com/MongoEngine/flask-mongoengine/pull/527
https://github.com/MongoEngine/flask-mongoengine/pull/528

**Pull request closed without merging**